wazimap_ng/cache.py

```diff
@@ -65,55 +72,86 @@ def last_modified_point_updated(request, profile_id, profile_category_id=None, t
 
     return last_modified(request, profile_id, key)
 
+
 ########### Signals #################
 def update_profile_cache(profile):
     logger.info(f"Updating profile cache: {profile}")
     key = profile_key % profile.id
     cache.set(key, datetime.now())
 
+
 def update_point_cache(category):
     profile = category.profile
     key1 = location_key % (profile.id, category.id)
 
     logger.debug(f"Set cache key (category): {key1}")
     cache.set(key1, datetime.now())
 
+
 @receiver(post_save, sender=ProfileIndicator)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
+
 @receiver(post_save, sender=ProfileHighlight)
 def profile_highlight_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
+
 @receiver(post_save, sender=IndicatorCategory)
 def profile_category_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
+
 @receiver(post_save, sender=IndicatorSubcategory)
 def profile_subcategory_updated(sender, instance, **kwargs):
     update_profile_cache(instance.category.profile)
 
+
 @receiver(post_save, sender=ProfileKeyMetrics)
 def profile_keymetrics_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
+
 @receiver(post_save, sender=Profile)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance)
 
+
 @receiver(post_save, sender=Location)
 def point_updated_location(sender, instance, **kwargs):
     update_point_cache(instance.category)
 
+
 @receiver(post_save, sender=Category)
 def point_updated_category(sender, instance, **kwargs):
     update_point_cache(instance)
 
+
+@receiver(post_save, sender=Indicator)
+def indicator_updated(sender, instance, **kwargs):
+    indicator_id = instance.id
+    all_profile_indicators = ProfileIndicator.objects.filter(
+        indicator_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    all_profile_key_metrics = ProfileKeyMetrics.objects.filter(
+        variable_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    all_profile_highlights = ProfileHighlight.objects.filter(
+        indicator_id=indicator_id
+    ).values_list('profile_id', flat=True)
+    # find the unique profiles whose cache needs to be invalidated
+    unique_profile_ids_to_invalidate = set(list(all_profile_indicators) + list(all_profile_key_metrics) + list(all_profile_highlights))
```

Comment:
so it seems the test has prevented a bug :) which is awesome :) 
we could use a union here, right? might be cleaner than the set and list calls. 
what do you think? 
`unique_profile_ids_to_invalidate = all_profile_indicators.union(all_profile_key_metrics, all_profile_highlights)`

Reply:
Do you prefer `.union` vs the `|` operator?

Reply:
I would prefer `|` but it does not work here.

```
AssertionError: Cannot combine queries on two different base models.
```

Reply:
I was thinking of set unions not queryset unions, but it's minor and probably just a matter of taste